src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -569,7 +569,12 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         assert Thread.holdsLock(this);
         ensureOpen();
         settings = getSettings(nodeId, seed, settings);
-        String name = buildNodeName(nodeId);
+        String name;
+        if (settings.get("node.name") != null) {
```

Comment:
yes I really want this to be predictable and deterministic. we have to prevent arbitrary naming here IMO